packages/next/src/shared/lib/router/utils/is-bot.ts

```diff
@@ -1,6 +1,19 @@
-const BOT_UA_RE =
-  /Googlebot|Mediapartners-Google|AdsBot-Google|googleweblight|Storebot-Google|Google-PageRenderer|Bingbot|BingPreview|Slurp|DuckDuckBot|baiduspider|yandex|sogou|LinkedInBot|bitlybot|tumblr|vkShare|quora link preview|facebookexternalhit|facebookcatalog|Twitterbot|applebot|redditbot|Slackbot|Discordbot|WhatsApp|SkypeUriPreview|ia_archiver/i
+// Bot crawler that will spin up a headless browser
+const HEADLESS_BOT_UA_RE =
+  /Googlebot|Google-PageRenderer|AdsBot-Google|googleweblight|Storebot-Google|Bingbot|BingPreview|applebot|facebookexternalhit|facebookcatalog|Twitterbot|LinkedInBot|Slackbot|Discordbot|WhatsApp|SkypeUriPreview/i
+
+// Static fetcher bot crawler that doesn't spin up a headless browser or execute JS
+const STATIC_FETCHER_BOT_UA_RE =
+  /Mediapartners-Google|Slurp|DuckDuckBot|baiduspider|yandex|sogou|bitlybot|tumblr|vkShare|quora link preview|redditbot|ia_archiver/i
+
+function isHeadlessBrowserBot(userAgent: string) {
+  return HEADLESS_BOT_UA_RE.test(userAgent)
+}
+
+function isNonBrowserBot(userAgent: string) {
```

Comment:
ultimately I think we want this to support a user-provided regex but if you're planning to do that in a future PR that's fine

Reply:
through nextjs config? for this feature or sth more general?

Reply:
yeah, this would be in a next config, so the user can decide which user agents should be classified as html-limited bots (in case we don't identify them all, or there are new ones that we aren't aware of and people don't want to wait for us to release a new version of Next.js to change it)